test/utils/ktesting/examples/ginkgo/example_test.go

```diff
@@ -0,0 +1,71 @@
+//go:build example
+// +build example
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ginkgo_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
+	"k8s.io/kubernetes/test/utils/ktesting"
+)
+
+var _ = ginkgo.Describe("something", func() {
```

Comment:
I'm personally not fan of creating more abstractions on tests and I will not be in favor of moving the sig network tests to this model, no opinion on what other SIGs should do

Reply:
I think this is harder to follow, did we discuss this anywhere?

Reply:
it is more "go like" to be explicit and avoid over-abstracting.

Now to read this code I need to know ginkgo and this ktesting context, I don't see why a helper for creating objects requires creating another context type.

Also, context should really only be used for timeout and tracing, it's a slow (WithValue is a linked-list-of-function-calls) and difficult to reason about (because by design it smuggles additional arbitrary types/data into functions) and should be used sparingly aside from cancellation, we should not lean heavier into hiding complexity inside context.Context.

Reply:
The net win for the Kubernetes project as a whole is that instead different SIGs maintaining their own "load and create object from YAML manifests" code we now only need one such implementation.

Reply:
> I think this is harder to follow, did we discuss this anywhere?

I've mentioned it a few times, but aside that from that is has been a side project of mine that hasn't been discussed much.

> hiding complexity inside context.Context

ktesting.TContext is more than that. It's conceptually more like the "test context" structs that we already have in various places.

The main difference is that it works for Gingko *and* Go test suites.

This one example is not a good place to discuss this concept. At some point I can prepare a presentation for a SIG Scheduling meeting or the contributor summit and we can discuss it further.

If I can't get this merged, than I will have to extend https://github.com/kubernetes/kubernetes/blob/bcadbfcc55ae591ec828600233a6fe826a7f78b4/test/e2e/storage/utils/create.go#L257-L273 (does not support validating admission policy at the moment) or copy the code from https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler_perf/create.go (supports any types, including those for VAP, but cannot be used in E2E tests).

Reply:
@jsafrane, @sanposhiho, @kerthcet: as users of the current code which creates objects from YAML files, what do you think? Is it worthwhile to consolidate that code in one place? What should the API for that code be?

Reply:
@jsafrane: In the meantime, is it okay if I extend https://github.com/kubernetes/kubernetes/blob/bcadbfcc55ae591ec828600233a6fe826a7f78b4/test/e2e/storage/utils/create.go#L257-L273 with support for VAP? That'll be a separate PR.

Reply:
I didn't read through all things this PR is trying to do though, (only) regarding `kobject` package, creating objects from Yaml looks common enough to create one central package to do that (actually I often do this in other projects of mine too). So +1.

> What should the API for that code be?

The current implementation looks simple enough for me.